### ISSUE TITLE
Support some kind of "about to be destroyed" async notification.

### ISSUE THREAD

**Issue by tmontes** [tmontes/ppytty#123]
Idea:
* Let a given task know that it is about to be destroyed.
* It would allow it, for example, to (try to) terminate spawned processes and wait on them.

Challenges:
* Such "callbacks" would actually be other tasks and be scheduled by the kernel.
* The about-to-be-destoyed task would have to be put on hold until such "callback" terminates.

Implementing #34 in full, may depend on this.
